connect/runtime/src/main/java/org/apache/kafka/connect/runtime/RestartRequest.java

```diff
@@ -0,0 +1,147 @@
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.runtime;
+
+import java.util.Objects;
+
+import org.apache.kafka.connect.connector.Connector;
+import org.apache.kafka.connect.connector.Task;
+
+/**
+ * A request to restart a connector and/or task instances.
+ * <p>The natural order is based first upon the connector name and then requested restart behaviors. 
+ * If two requests have the same connector name, then the requests are ordered based on the 
+ * probable number of tasks/connector this request is going to restart.
+ */
+public class RestartRequest implements Comparable<RestartRequest> {
+
+    private final String connectorName;
+    private final boolean onlyFailed;
+    private final boolean includeTasks;
+
+    /**
+     * Create a new request to restart a connector and optionally its tasks.
+     *
+     * @param connectorName the name of the connector; may not be null
+     * @param onlyFailed    true if only failed instances should be restarted
+     * @param includeTasks  true if tasks should be restarted, or false if only the connector should be restarted
+     */
+    public RestartRequest(String connectorName, boolean onlyFailed, boolean includeTasks) {
+        this.connectorName = Objects.requireNonNull(connectorName, "Connector name may not be null");
+        this.onlyFailed = onlyFailed;
+        this.includeTasks = includeTasks;
+    }
+
+    /**
+     * Get the name of the connector.
+     *
+     * @return the connector name; never null
+     */
+    public String connectorName() {
+        return connectorName;
+    }
+
+    /**
+     * Determine whether only failed instances be restarted.
+     *
+     * @return true if only failed instances should be restarted, or false if all applicable instances should be restarted
+     */
+    public boolean onlyFailed() {
+        return onlyFailed;
+    }
+
+    /**
+     * Determine whether {@link Task} instances should also be restarted in addition to the {@link Connector} instance.
+     *
+     * @return true if the connector and task instances should be restarted, or false if just the connector should be restarted
+     */
+    public boolean includeTasks() {
+        return includeTasks;
+    }
+
+    /**
+     * Determine whether the connector with the given status is to be restarted.
+     *
+     * @param status the connector status; may not be null
+     * @return true if the connector is to be restarted, or false otherwise
+     */
+    public boolean shouldRestartConnector(ConnectorStatus status) {
+        return !onlyFailed || status.state() == AbstractStatus.State.FAILED;
+    }
+
+    /**
+     * Determine whether only the {@link Connector} instance is to be restarted even if not failed.
+     *
+     * @return true if only the {@link Connector} instance is to be restarted even if not failed, or false otherwise
+     */
+    public boolean forciblyRestartConnectorOnly() {
```

Comment:
```suggestion
    public boolean forceRestartConnectorOnly() {
```
? (seems a bit more common way to say it, but I don't have a strong opinion)